lib/index.js

```diff
@@ -0,0 +1,295 @@
+				this._getSchemaPathsList(nodes[key], schemaPathsList);
+		}
+
+		return schemaPathsList;
+	}
+
+	/**
+	 * Parses the content of a file
+	 * @param {string} fileType The file type extension
+	 * @param {string} fileContent The string with the file content
+	 */
+	_parseFile(fileType, fileContent) {
+		switch(fileType) {
+			case 'yml':
+			case 'yaml':
+				try {
+					return YAML.parse(fileContent);
+				} catch(error) {
+					throw new SyntaxError(`${error.message} Line ${error.parsedLine}.`);
+				}
+			case 'json':
+				return JSON.parse(fileContent);
+			default:
+				return null;
+		}
+	}
+
+	/**
+	 * Retrieves the content of the schema file paths
+	 * @async
+	 * @param {Array<string>} schemaPaths The list of schemas file paths
+	 * @return {Array{object}}
+	 */
+	_readSchemaFiles(schemaPaths) {
+		return Promise.all(schemaPaths.map(async pathname => {
+			const fileType = pathname.split('.').pop();
+			const file = await fs.readFile(pathname, 'utf8');
+
+			try {
+				const parsedFile = this._parseFile(fileType, file);
+				return parsedFile;
+			} catch(error) {
+				error.message = `Invalid file ${pathname}, not a valid ${fileType.toUpperCase()}. ${error.message}`;
+				throw error;
+			}
+		}));
+	}
+
+	/**
+	 * Merges the schemas into a single object
+	 * @param {Array<Object>} schemas The list of schemas parts to merge
+	 */
+	_mergeSchemas(schemas) {
+		const mergedSchemas = {};
+
+		merge(mergedSchemas, ...schemas);
+
+		return mergedSchemas;
+	}
+
+	/**
+	 * Validates if a schema is compliant with the specification
+	 * @async
+	 * @param {string} schemaType The type of the schema like "public"
+	 * @param {Object} schema The final schema object
+	 */
+	async _validateSchema(schemaType, schema) {
+
+		const schemaValidator = new OpenapiSchemaValidator({ version: 3 });
+		const results = schemaValidator.validate(schema);
+
+		if(!results.errors.length)
+			return;
+		// If Schemas has an error
+		try {
+			await fs.writeFile(this.constructor.errorLog, JSON.stringify({ [schemaType]: results }, null, '\t'));
+		} catch(error) {
+			// if can't make a file
+			throw new Error(`Found errors on schema of type "${schemaType}". Can't Write build-error.log file `);
+		}
+
+		throw new Error(`Found errors on schema of type "${schemaType}". Check build-error.log file for more details`);
+	}
+
+	/**
+	 * Builds the schema by type and writes the JSON file in the buildPath
+	 * @param {string} schemaType The type of the schema like "public"
+	 * @param {Object} schemaTree The schema tree with the schema file paths and modules
+	 */
+	async _buildSchema(schemaType, schemaTree) {
+		const schemaPaths = this._getSchemaPathsList(schemaTree);
+		const schemaObjects = await this._readSchemaFiles(schemaPaths);
+		const schema = this._mergeSchemas(schemaObjects);
+
+		const resolvedSchemaResult = await jsonRefs.resolveRefs(schema, {
+			filter: ['relative', 'remote'],
+			location: `${this.constructor.schemaDir}${schemaType}.json`
```

Comment:
Ojo acá: este `jsonRefs` resuelve los `$ref` de las APIs.
Pero si lo que estás referenciando es otro archivo yaml, lo único que hace es traerte texto.

Acá la sugerencia (a validar que funcione OK) de algo que tenía hecho en otro repo:

```suggestion
			location: `${this.constructor.schemaDir}${schemaType}.json`,
			loaderOptions: {
				processContent: (content, callback) => {

					if(content.location.match(/\.ya?ml$/i))
						return callback(undefined, YAML.parse(content.text));

					return callback(undefined, JSON.parse(content.text));

				}
			}
```

Ejemplo de `$ref` que rompería ahora mismo: `$ref: '../../some-other-schema.yml'`